scan2.py

```diff
@@ -365,13 +365,16 @@ def _for_partial_eval(trace, *tracers, jaxpr, nsteps, reverse):
   body_jaxpr, () = discharge_state(jaxpr, ())
   for _ in range(len(in_unknowns)):
     _, _, out_unknowns, _, num_res = \
-        pe._partial_eval_jaxpr_custom(body_jaxpr, [False, *in_unknowns], _save_anything)
+        pe._partial_eval_jaxpr_custom(body_jaxpr, [False, *in_unknowns],
+                                      _save_anything)
+    out_unknowns = list(out_unknowns)
     if out_unknowns == in_unknowns:
```

Comment:
```suggestion
    if list(out_unknowns) == in_unknowns:
```

Sorry, can't resist golfing :P